File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):  # inherit from Node
    def __init__(self):
        super().__init__("py_test")
        self.get_logger().info("Hello there!")
        self.create_timer(0.5, self.timer_callback)
        self.counter = 0
    
    def timer_callback(self):
        self.get_logger().info(str(self.counter))
        self.counter += 1

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode: public rclcpp::Node {
public:
    MyNode(): Node("cpp_test"), counter(0){
        RCLCPP_INFO(this->get_logger(), "Hello Cpp!");
        timer = this->create_wall_timer(std::chrono::seconds(1), // time
                                        std::bind(&MyNode::timerCallback, this)); // binduje funkcje
    }

private:
    void timerCallback(){
        RCLCPP_INFO(get_logger(), "Hello! %d", counter);
        counter++;
    }

    rclcpp::TimerBase::SharedPtr timer;
    uint16_t counter;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}